sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -700,31 +700,27 @@ public void close() {
      * entity.
      */
     private void queueWork(int maximumMessageCount, Duration maxWaitTime,
-                           FluxSink<ServiceBusReceivedMessage> emitter) {
+        Sinks.Many<ServiceBusReceivedMessage> emitter) {
+
         final long id = idGenerator.getAndIncrement();
-        final int prefetch = asyncClient.getReceiverOptions().getPrefetchCount();
-        final int toRequest = prefetch != 0 ? Math.min(maximumMessageCount, prefetch) : maximumMessageCount;
-        final SynchronousReceiveWork work = new SynchronousReceiveWork(id,
-            toRequest,
-            maxWaitTime, emitter);
-        SynchronousMessageSubscriber messageSubscriber = synchronousMessageSubscriber.get();
-        if (messageSubscriber == null) {
-            SynchronousMessageSubscriber newSubscriber = new SynchronousMessageSubscriber(toRequest, work);
-            if (!synchronousMessageSubscriber.compareAndSet(null, newSubscriber)) {
-                newSubscriber.dispose();
-                SynchronousMessageSubscriber existing = synchronousMessageSubscriber.get();
-                existing.queueWork(work);
-            } else {
-                asyncClient.receiveMessagesNoBackPressure().subscribeWith(newSubscriber);
-            }
-        } else {
+        final SynchronousReceiveWork work = new SynchronousReceiveWork(id, maximumMessageCount, maxWaitTime, emitter);
+        final SynchronousMessageSubscriber messageSubscriber = synchronousMessageSubscriber.get();
+
+        if (messageSubscriber != null) {
             messageSubscriber.queueWork(work);
+            return;
         }
-        logger.verbose("[{}] Receive request queued up.", work.getId());
-    }
 
-    OffsetDateTime renewSessionLock(String sessionId) {
-        return asyncClient.renewSessionLock(sessionId).block(operationTimeout);
+        final SynchronousMessageSubscriber newSubscriber = new SynchronousMessageSubscriber(work);
+
+        if (synchronousMessageSubscriber.compareAndSet(null, newSubscriber)) {
+            asyncClient.receiveMessagesNoBackPressure().subscribeWith(newSubscriber);
```

Comment:
Hi @conniey, Is the idea to asynchronously send the credit to the service as soon as receiveMessage() API is called (first time)? meaning by the time application start iterating the iterable, there could be already some messages buffered in the emitter serving the messages for iteration. I just want to confirm, if so, we could document it so that users know calling this method can increment the delivery count (irrespective of they iterate or not)

Reply:
Ah yes. That's worthwhile to document. Good to point out.